test/integration/mqtt_paho_binding/mqtt_test.go

```diff
@@ -0,0 +1,99 @@
+package mqtt_paho_binding
+
+import (
+	"context"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/eclipse/paho.golang/paho"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cloudevents/sdk-go/protocol/mqtt_paho/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/event"
+	bindings "github.com/cloudevents/sdk-go/v2/protocol"
+	"github.com/cloudevents/sdk-go/v2/protocol/test"
+
+	. "github.com/cloudevents/sdk-go/v2/binding/test"
+	. "github.com/cloudevents/sdk-go/v2/test"
+)
+
+func TestSendStructuredMessageToStructured(t *testing.T) {
+	ctx := context.Background()
+	s, r, close := testSenderReceiver(t, ctx)
+	defer close()
+
+	EachEvent(t, Events(), func(t *testing.T, eventIn event.Event) {
+		eventIn = ConvertEventExtensionsToString(t, eventIn)
+		in := MustCreateMockStructuredMessage(t, eventIn)
+
+		test.SendReceive(t, binding.WithPreferredEventEncoding(ctx, binding.EncodingStructured), in, s, r,
+			func(out binding.Message) {
+				eventOut := MustToEvent(t, ctx, out)
+				assert.Equal(t, binding.EncodingStructured, out.ReadEncoding())
+				AssertEventEquals(t, eventIn, ConvertEventExtensionsToString(t, eventOut))
+			})
+	})
+}
+
+func TestSendBinaryMessageToBinary(t *testing.T) {
+	ctx := context.Background()
+	s, r, close := testSenderReceiver(t, ctx)
+	defer close()
+
+	EachEvent(t, Events(), func(t *testing.T, eventIn event.Event) {
+		eventIn = ConvertEventExtensionsToString(t, eventIn)
+		in := MustCreateMockBinaryMessage(eventIn)
+		test.SendReceive(t, binding.WithPreferredEventEncoding(ctx, binding.EncodingBinary), in, s, r,
+			func(out binding.Message) {
+				eventOut := MustToEvent(t, ctx, out)
+				assert.Equal(t, binding.EncodingBinary, out.ReadEncoding())
+				eventOut = ConvertEventExtensionsToString(t, eventOut)
+
+				AssertEventEquals(t, eventIn, ConvertEventExtensionsToString(t, eventOut))
+			})
+	})
+}
+
+func testSenderReceiver(t testing.TB, ctx context.Context) (bindings.Sender, bindings.Receiver, func()) {
```

Comment:
lower case test?

Reply:
Sorry for the unclear naming. It's only for getting the sender and receiver, so change it to `getSenderReceiver`.